### PR TITLE
[F13]: add baseRequest., messageField

### PR DESCRIPTION
**Pull request by nphumy98**
Add Validation for Login. need to approve both changes in Fr8model and Fr8back

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -4,8 +4,8 @@
         "module": "es6",
         "declaration": true,
         "outDir": "./dist",
-        "strict": true
-
+        "strict": true,
+        "strictPropertyInitialization": false
```
Comment (chownema) [inline tsconfig.json]: Could you please explain why?

File: src/model/utils/messageField.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export class MessageField {
+    public field: string;
+    public reason: string;
+
+    constructor(field: string, reason: string) {
+        this.field = field;
+        this.reason= reason;
```
Comment (chownema) [inline src/model/utils/messageField.ts]: Space

File: src/model/utils/errorResponse.ts [hunk 3]
```diff
@@ -1,3 +1,11 @@
+import { MessageField } from "./messageField";
+
 export class ErrorResponse {
-    constructor(public error?: string, public message?: string){}
+    public error: string;
```
Comment (chownema) [inline src/model/utils/errorResponse.ts]: Why would you want to remove constructors?

File: src/model/request/loginRequest.ts [hunk 4]
```diff
@@ -1,3 +1,22 @@
-export class LoginRequest {
-    constructor(public username?: string, public password?: string){ }
```
Comment (chownema) [inline src/model/request/loginRequest.ts:2]: Why did you remove the constructor?

**Review by chownema — CHANGES_REQUESTED** [review]

File: tsconfig.json (hunk 1 above)
Comment (nphumy98) [reply tsconfig.json]: That is weird, 3 things you mention make the project didn't compile before. now it work. must be a monkey. I changed it back

File: src/model/utils/errorResponse.ts (hunk 3 above)
Comment (nphumy98) [reply src/model/utils/errorResponse.ts]: This thing make the Fre8back didnt compile before. Dont know why it work now. Fixed . Changed it back

File: src/model/request/loginRequest.ts (hunk 4 above)
Comment (nphumy98) [reply src/model/request/loginRequest.ts:2]: This thing make the Fre8back didnt compile before. Dont know why it work now. Fixed . Changed it back

File: tsconfig.json (hunk 1 above)
Comment (chownema) [reply tsconfig.json]: Sorry could you clarify what you did to build which resulted in the project not building?
